lib.php

```diff
@@ -440,6 +440,22 @@ public function enrol_self(stdClass $instance, $user = null, $charge = true) {
             do {
                 $this->enrol_user($instance, $user->id, $instance->roleid, $timestart, $timeend, null, true);
             } while (!$DB->record_exists('user_enrolments', $conditions));
+
+            try {
```

Comment:
The event already should be triggered  by  enrol_user method, Why trigger it again?